### DISCUSSION TITLE
How do I run the seeder files?

### DISCUSSION THREAD

**Discussion by tvarga94** [austintoddj/canvas#1425] [General]
Hello, is there a way to run the seeder files included in the package?

in the database/factories directory there are Factories made for the tables created by the package. Is there a way to run them? 
(The factories are not classes and the Model does not have the HasFactory trait included)

Thank you for the answers in advance,

**Comment by tvarga94** [comment]
@austintoddj maybe I missed it in the documentation?

**Comment by austintoddj** [comment]
@tvarga94 The factories in Canvas are used for the tests. Seeder files aren't included with the package.

**Reply by tvarga94** [reply]
oh, okay, I misunderstood their usage, thank you for the clarification and for the answer

**Reply by austintoddj** [reply]
You're welcome. [This guide can help with creating View & Visits](https://github.com/austintoddj/canvas/blob/master/.github/CONTRIBUTING.md#installation) when you're developing, you could use the same concept for creating Tags & Topics.

**Comment by austintoddj** [comment]
Closing with a v7 clarification:

Package factories under `database/factories` are for **Canvas’s own test suite**, not a public “seed demo content” product. There isn’t a first-party Artisan seeder you run in the host app to populate a blog.

For local/demo data, create posts in `/canvas` or write a small host seeder that uses the `Canvas\Models\*` models (and grants access with `canvas:make-admin` / `canvas:assign-role`).
